cmd/libsnap-confine-private/tool.c

```diff
@@ -117,6 +117,14 @@ void sc_call_snap_update_ns_as_user(int snap_update_ns_fd,
 				 "XDG_RUNTIME_DIR=%s", xdg_runtime_dir);
 	}
 
+	const char *snap_real_home = getenv("SNAP_REAL_HOME");
```

Comment:
I know this might be a bit naive, but why do we get SNAP_REAL_HOME from env and not calculate the home dir for the uid inside snap-confine directly? when used previously this was fine because only the "confined" snap uses the env, but now that snap-update-ns is using it, it could get a little tricky.

Reply:
Thanks for the overview